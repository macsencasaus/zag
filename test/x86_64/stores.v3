// RUN: %vector -stdout %s > /tmp/tmp.bin && objdump -d /tmp/tmp.bin

fn main(): i32 {
    var x: i32 = 2;
    return x;
}

fn big_stack(): i32 {
    var a: i64,
        b: i64,
        c: i64,
        d: i64,
        e: i64,
        f: i64,
        g: i64,
        h: i64,
        i: i64,
        j: i64,
        k: i64,
        l: i64,
        m: i64,
        n: i64;
    return 0;
}

fn params(a: i32, b: i32): i32 {
    return a;
}

fn add(a: i32, b: i32): i32 {
    return a + b;
}
