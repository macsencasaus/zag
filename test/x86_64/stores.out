
/tmp/tmp.bin:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <main>:
   0:	55                   	push   rbp
   1:	48 89 e5             	mov    rbp,rsp
   4:	48 83 ec 10          	sub    rsp,0x10
   8:	b8 02 00 00 00       	mov    eax,0x2
   d:	89 45 fc             	mov    DWORD PTR [rbp-0x4],eax
  10:	8b 45 fc             	mov    eax,DWORD PTR [rbp-0x4]
  13:	c9                   	leave
  14:	c3                   	ret

0000000000000015 <big_stack>:
  15:	55                   	push   rbp
  16:	48 89 e5             	mov    rbp,rsp
  19:	48 83 ec 70          	sub    rsp,0x70
  1d:	b8 00 00 00 00       	mov    eax,0x0
  22:	c9                   	leave
  23:	c3                   	ret

0000000000000024 <params>:
  24:	55                   	push   rbp
  25:	48 89 e5             	mov    rbp,rsp
  28:	48 83 ec 10          	sub    rsp,0x10
  2c:	89 7d fc             	mov    DWORD PTR [rbp-0x4],edi
  2f:	89 75 f8             	mov    DWORD PTR [rbp-0x8],esi
  32:	8b 45 fc             	mov    eax,DWORD PTR [rbp-0x4]
  35:	c9                   	leave
  36:	c3                   	ret

0000000000000037 <add>:
  37:	55                   	push   rbp
  38:	48 89 e5             	mov    rbp,rsp
  3b:	48 83 ec 10          	sub    rsp,0x10
  3f:	89 7d fc             	mov    DWORD PTR [rbp-0x4],edi
  42:	89 75 f8             	mov    DWORD PTR [rbp-0x8],esi
  45:	8b 45 fc             	mov    eax,DWORD PTR [rbp-0x4]
  48:	8b 4d f8             	mov    ecx,DWORD PTR [rbp-0x8]
  4b:	01 c8                	add    eax,ecx
  4d:	89 45 f4             	mov    DWORD PTR [rbp-0xc],eax
  50:	8b 45 f4             	mov    eax,DWORD PTR [rbp-0xc]
  53:	c9                   	leave
  54:	c3                   	ret

0000000000000055 <store>:
  55:	55                   	push   rbp
  56:	48 89 e5             	mov    rbp,rsp
  59:	48 83 ec 20          	sub    rsp,0x20
  5d:	b8 02 00 00 00       	mov    eax,0x2
  62:	48 89 45 f8          	mov    QWORD PTR [rbp-0x8],rax
  66:	48 8d 45 f8          	lea    rax,[rbp-0x8]
  6a:	48 89 45 f0          	mov    QWORD PTR [rbp-0x10],rax
  6e:	48 8b 45 f0          	mov    rax,QWORD PTR [rbp-0x10]
  72:	48 89 45 f0          	mov    QWORD PTR [rbp-0x10],rax
  76:	48 8b 45 f0          	mov    rax,QWORD PTR [rbp-0x10]
  7a:	48 89 45 e8          	mov    QWORD PTR [rbp-0x18],rax
  7e:	b8 03 00 00 00       	mov    eax,0x3
  83:	48 8b 4d e8          	mov    rcx,QWORD PTR [rbp-0x18]
  87:	48 89 01             	mov    QWORD PTR [rcx],rax
  8a:	48 8b 45 f0          	mov    rax,QWORD PTR [rbp-0x10]
  8e:	48 89 45 e8          	mov    QWORD PTR [rbp-0x18],rax
  92:	48 8b 45 e8          	mov    rax,QWORD PTR [rbp-0x18]
  96:	48 8b 00             	mov    rax,QWORD PTR [rax]
  99:	c9                   	leave
  9a:	c3                   	ret
